flags/settings.py

```diff
@@ -0,0 +1,43 @@
+import logging
+
+from importlib import import_module
+
+from django.conf import settings
+
+logger = logging.getLogger(__name__)
+
+# Private set of all global flags from sources in FLAG_SOURCES. This is
+# populated on AppConfig.ready() by add_flags_from_sources below.
+SOURCED_FLAGS = {}
+
+
+def add_flags_from_sources(sources=None):
+    """ Read flags from sources defined in settings.FLAG_SOURCES.
+    FLAG_SOURCES is expected to be a list of Python module with flags
+    specified with variable assignment, e.g. MY_FLAG = True """
+    global SOURCED_FLAGS
+
+    if sources is None:
+        sources = getattr(settings, 'FLAG_SOURCES', ())
+
+    for source_str in sources:
+        source = import_module(source_str)
+        for flag in (f for f in dir(source)
+                     if f.isupper() and isinstance(getattr(source, f), bool)):
+            SOURCED_FLAGS[flag] = getattr(source, flag)
+
+
+def get_global_flags(sourced_flags=None):
+    """ Get all global flags and their state.
+    This combines FLAGS from settings with all possible FLAG_SOURCES. """
+    global_flags = getattr(settings, 'FLAGS', {})
+    if sourced_flags is None:
+        sourced_flags = SOURCED_FLAGS
+
+    # Global flags override everything. If there is a flag in a module defined
+    # in settings.FLAG_SOURCES and a flag defined in settings.FLAGS,
+    # settings.FLAGS wins.
```

Comment:
Edited the above comment: I think in this case raising an exception doesn't make sense — in this case a global flag could override a database flag. It might be worth adding a warning to the admin template when there are overlapping flags and logging a warning. But raising an exception here will cause the site to 500.